src/Faithlife.WebRequests/Json/ExplicitStatusResponse.cs

```diff
@@ -0,0 +1,32 @@
+using System.Net;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Faithlife.WebRequests.Json
+{
+	/// <summary>
+	/// AutoWebServiceRequest responses with an explicitly set status code. 
+	/// </summary>
+	public abstract class ExplicitStatusResponse : AutoWebServiceResponse
+	{
+		/// <summary>
+		/// Exposes the response status code.
+		/// </summary>
+		public HttpStatusCode StatusCode { get; internal set; } // internal instead of private to avoid being compiled away
+
+		/// <summary>
+		/// Exposes the response headers.
+		/// </summary>
+		public HttpResponseHeaders Headers { get; private set; }
```

Comment:
Perhaps we should support this property directly, like `StatusCode`, so that clients don't need to override `OnResponseHandledCoreAsync`.

Reply:
> Perhaps we should support this property directly

We could, easily, but this minimizes the amount of reflection magic being done.

> so that clients don't need to override OnResponseHandledCoreAsync.

I don't understand this point yet; having this override doesn't impose any need upon this class' clients, and existing `AutoWebServiceResponse`'s clients already have the option to override it if they want.

Reply:
It just seemed odd to use reflection for `StatusCode` but `OnResponseHandledCoreAsync` for `Headers`. Other clients that don't use this base class might like that behavior for headers.

Also, it would be nice if derived classes didn't have to know to call `base.OnResponseHandledCoreAsync`.

Reply:
>It just seemed odd to use reflection for StatusCode but OnResponseHandledCoreAsync for Headers

Agreed it's a bit odd, but there's precedent in the current `AutoWebServiceResponse` class and I do also like minimizing magic.

> Other clients that don't use this base class might like that behavior for headers.

And they can get it, both before and after this change, by overriding this method as this class does.

> it would be nice if derived classes didn't have to know to call `base.OnResponseHandledCoreAsync`.

They currently already need to know to do that; see all the stuff the base function this one calls does: 
https://github.com/Faithlife/FaithlifeWebRequests/blob/91c612dd567044bfdbce1ce10ec9cf1e5cf87e9a/src/Faithlife.WebRequests/Json/AutoWebServiceResponse.cs#L42-L56

Reply:
> They currently already need to know to do that

Whoa, good point.

Reply:
Left as is per discussion. 9c6a8a47ed60beacfa362fcb6831fa078b832668

Reply:
I had a similar objection in that `StatusCode` is magically set but `Headers` has to be manually set if not using the `GenericStatusCodeResponse` base class.  That said, we didn't see a clean way to magically set both.  We could consider only setting `StatusCode` if the base class is used so that `Headers` and `StatusCode` are consistently populated, but then callers would _have_ to derive from the `GenericStatusCodeResponse`.